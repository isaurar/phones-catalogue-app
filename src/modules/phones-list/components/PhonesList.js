import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {
    List,
    ListItem,
    ListItemIcon,
    Divider
} from '@material-ui/core';
import IconAndroidPhone from '@material-ui/icons/PhoneAndroid';
import {useHistory} from 'react-router-dom';

const PhonesList = ({data}) => {
    const history = useHistory();
    return data ? <div>
        <List dense>
            {data.map(({id, title}, i) => <Fragment  key={i}>
                <ListItem onClick={() => history.push(`/phone-details/${id}`)}>
                    <ListItemIcon>
                        <IconAndroidPhone />
                    </ListItemIcon>
                        #{id} {title}
                </ListItem>
                <Divider />
            </Fragment>)}
        </List>
    </div> : 'Empty';
};

PhonesList.propTypes = {
    data: PropTypes.array
};

export default PhonesList