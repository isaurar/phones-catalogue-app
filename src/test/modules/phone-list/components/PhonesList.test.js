import React from 'react';
import TestRenderer from 'react-test-renderer';
import PhonesList from '../../../../modules/phones-list/components/PhonesList.js';
import PhonesAPI from '../../../testData/testDataPhonesAPI.js';

const testData = PhonesAPI.getPhonesAndMetadataFiltered(1, 5).phones;

describe('phone list component', () => {
  test('should render the phones list correctly', () => {
    const rendered = TestRenderer.create(<PhonesList data={testData} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
}); 