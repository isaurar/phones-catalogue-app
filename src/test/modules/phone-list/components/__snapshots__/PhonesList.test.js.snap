// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phone list component should render the phones list correctly 1`] = `
<div>
  <ul
    className="MuiList-root MuiList-dense MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Apple iPhone 11
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Apple iPhone 11 Pro
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Apple iPhone 11 Pro Max
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Huawei P30
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-dense MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-dense"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
        >
          Huawei P30 lite
        </span>
      </div>
    </div>
    <hr
      className="MuiDivider-root"
    />
  </ul>
</div>
`;
